boot/assets.go

```diff
@@ -794,20 +807,29 @@ func observeSuccessfulBootAssetsForBootloader(m *Modeenv, root string, opts *boo
 	}
 
 	cache := newTrustedAssetsCache(dirs.SnapBootAssetsDir)
-	for _, trustedAsset := range trustedAssets {
-		assetName := filepath.Base(trustedAsset)
 
-		// find the hash of the file on disk
-		assetHash, err := cache.fileHash(filepath.Join(root, trustedAsset))
-		if err != nil && !os.IsNotExist(err) {
-			return nil, fmt.Errorf("cannot calculate the digest of existing trusted asset: %v", err)
+	alreadySeenAssetNames := make(map[string]bool)
+	for trustedAsset, assetName := range trustedAssets {
+		_, alreadySeen := alreadySeenAssetNames[assetName]
+		if alreadySeen {
+			// TrustedAssetsBootloader.TrustedAssets
+			// should not map different paths to the same
+			// name. If it does it is a bug.
+			logger.Noticef("bootloader %s has several asset of the same name %s", whichBootloader, assetName)
```

Comment:
I wonder as we are very much in control of this from the bootloader side, whether this should be an actual error?  cc @alfonsosanchezbeato @bboozzoo

Reply:
Probably this should return as an internal error, I agree